### ISSUE TITLE
Permission Denied

### ISSUE THREAD

**Issue by abhinavagrawal1995** [angular-schule/angular-cli-ghpages#29]
I am getting the following error-
`sudo ngh --branch=master --no-silent
Cloning git@github.com:abhinavagrawal1995/web-portfolio-angular.git into ../../../../usr/lib/node_modules/angular-cli-ghpages/node_modules/gh-pages/.cache

An error occurred!
 { Error
    at ChildProcess.<anonymous> (/usr/lib/node_modules/angular-cli-ghpages/node_modules/gh-pages/lib/git.js:47:23)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at maybeClose (internal/child_process.js:899:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
  code: 128,
  message: 'Cloning into \'../../../../usr/lib/node_modules/angular-cli-ghpages/node_modules/gh-pages/.cache\'...\nPermission denied (publickey).\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n',
  name: 'ProcessError' }`

I have verified that the repo is correct.

**Comment by JohannesHoppe** [comment]
> Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

You have no rights to push to the repo. In most cases this  belongs to the fact, that build agents (like travis-ci) have only the right to read, not to write. Please refer the the [README.md](https://github.com/angular-schule/angular-cli-ghpages/blob/master/README.md) and try out the GH_TOKEN feature! Cheers!

**Comment by carlosdelfino** [comment]
I Get the same problem, but in my case I use a pair of private public keys to access GitHub, using Git manually I do not have any problem, however it seems that the tool does not allow git to request the password of the private key, then the error occurs, preventing authentication next to GitHub.
```
$ ngh  --dir dist/webservices/ -S
Cloning git@github.com:islamic-works/webservices.git into ..\..\..\AppData\Roaming\npm\node_modules\angular-cli-ghpages\node_modules\gh-pages\.cache

*** An error occurred!

{ ProcessError: Cloning into '..\..\..\AppData\Roaming\npm\node_modules\angular-cli-ghpages\node_modules\gh-pages\.cache'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

    at ChildProcess.<anonymous> (C:\Users\Admin\AppData\Roaming\npm\node_modules\angular-cli-ghpages\node_modules\gh-pages\lib\git.js:47:23)
    at ChildProcess.emit (events.js:198:13)
    at maybeClose (internal/child_process.js:982:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
  code: 128,
  message:
   'Cloning into \'..\\..\\..\\AppData\\Roaming\\npm\\node_modules\\angular-cli-ghpages\\node_modules\\gh-pages\\.cache\'...\ngit@github.com: Permission denied (publickey).\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n',
  name: 'ProcessError' }
```

**Comment by Rblock919** [comment]
I realized this error is happening on my ubuntu machine and not my macbook so I think it has to do with how linux handles installing npm packages globally when you use sudo.

I ran `sudo chown -R $(whoami) /usr/local/lib/node_modules/angular-cli-ghpages/`, which probably isn't advised, and just using `ngh` works now. I figured I'll never run this command as root so I don't care if the directory belongs to my user anyhow.
